kore/src/Kore/Rewrite/Axiom/Identifier.hs

```diff
@@ -62,6 +63,9 @@ these can be identified.
 
 The expectation is that an axiom can be applied to a term only if the
 identifier of its left-hand-side is the same as the term's identifier.
+
+In practice, recursion should be cut short at a level that is
+selective enough for good performance (see 'matchAxiomIdentifier').
```

Comment:
:thinking: well... we may or may not want to add that (cutting the recursion short is not supported in the `recursion-schemes` flavour of `fold` AFAICT). Maybe I should delete this comment
```suggestion
```